patches/api/0274-Add-missing-effects.patch

```diff
@@ -28,49 +28,111 @@ index d0c812c72967469122f7f276ced31cfb0e9482dc..0f598fa0560d14b44dcc65205364870c
 +     */
++    @Deprecated
 +    BOOK_PAGE_TURNED(1043, Type.SOUND),
 +    /**
 +     * Particles displayed when a composter composts
++     *
++     * @deprecated use {@link #COMPOSTER_FILL_ATTEMPT}
 +     */
++    @Deprecated
 +    COMPOSTER_COMPOSTS(1500, Type.VISUAL),
 +    /**
 +     * Particles displayed when lava converts a block (either water to stone, or
 +     * removing existing blocks such as torches)
++     *
++     * @deprecated use {@link #LAVA_INTERACT}
 +     */
++    @Deprecated
 +    LAVA_CONVERTS_BLOCK(1501, Type.VISUAL),
 +    /**
 +     * Particles displayd when a redstone torch burns out
++     *
++     * @deprecated use {@link #REDSTONE_TORCH_BURNOUT}
 +     */
++    @Deprecated
 +    REDSTONE_TORCH_BURNS_OUT(1502, Type.VISUAL),
 +    /**
 +     * Particles displayed when an ender eye is placed
++     *
++     * @deprecated use {@link #END_PORTAL_FRAME_FILL}
 +     */
++    @Deprecated
 +    ENDER_EYE_PLACED(1503, Type.VISUAL),
 +    /**
 +     * Particles displayed when an ender dragon destroys block
++     *
++     * @deprecated use {@link #ENDER_DRAGON_DESTROY_BLOCK}
 +     */
++    @Deprecated
 +    ENDER_DRAGON_DESTROYS_BLOCK(2008, Type.VISUAL),
 +    /**
 +     * Particles displayed when a wet sponge vaporizes in nether.
++     *
++     * @deprecated use {@link #SPONGE_DRY}
 +     */
++    @Deprecated
 +    WET_SPONGE_VAPORIZES_IN_NETHER(2009, Type.VISUAL)
-+    // Paper end
      ;
++    private static final java.util.Set<Effect> DEPRECATED_PAPER_TYPES = java.util.EnumSet.of(PHANTOM_BITES, ZOMBIE_CONVERTS_TO_DROWNED, HUSK_CONVERTS_TO_ZOMBIE, GRINDSTONE_USED, BOOK_PAGE_TURNED, COMPOSTER_COMPOSTS, LAVA_CONVERTS_BLOCK, REDSTONE_TORCH_BURNS_OUT, ENDER_EYE_PLACED, ENDER_DRAGON_DESTROYS_BLOCK, WET_SPONGE_VAPORIZES_IN_NETHER);
++    // Paper end
  
      private final int id;
+     private final Type type;
+@@ -367,6 +458,7 @@ public enum Effect {
+ 
+     static {
+         for (Effect effect : values()) {
++            if (!DEPRECATED_PAPER_TYPES.contains(effect)) // Paper
+             BY_ID.put(effect.id, effect);
+         }
+     }
+diff --git a/src/test/java/org/bukkit/EffectTest.java b/src/test/java/org/bukkit/EffectTest.java
+index 54e621e86e8fe3414099494d419929b282b33489..cfdea0c853592e0daf8a9cc2c46fe8bf7f2b6984 100644
+--- a/src/test/java/org/bukkit/EffectTest.java
++++ b/src/test/java/org/bukkit/EffectTest.java
+@@ -1,13 +1,16 @@
+ package org.bukkit;
+ 
++import static org.bukkit.Effect.*; // Paper
+ import static org.hamcrest.CoreMatchers.*;
+ import static org.junit.Assert.*;
+ import org.junit.Test;
+ 
+ public class EffectTest {
++    private static final java.util.Set<Effect> DEPRECATED_PAPER_TYPES = java.util.EnumSet.of(PHANTOM_BITES, ZOMBIE_CONVERTS_TO_DROWNED, HUSK_CONVERTS_TO_ZOMBIE, GRINDSTONE_USED, BOOK_PAGE_TURNED, COMPOSTER_COMPOSTS, LAVA_CONVERTS_BLOCK, REDSTONE_TORCH_BURNS_OUT, ENDER_EYE_PLACED, ENDER_DRAGON_DESTROYS_BLOCK, WET_SPONGE_VAPORIZES_IN_NETHER); // Paper
```

Comment:
Could probably either detect Deprecated or reflect if you disliked this

Reply:
Not pretty, but at least slightly nicer!